src/lib.rs

```diff
@@ -300,41 +330,95 @@ impl Analyzer {
+                    .clone()
+                    .into_iter()
+                    .find(|x| import_path.string.starts_with(&x.name));
+
+                let remapped = if remapping.is_some() {
+                    let unwrapped = remapping.clone().unwrap();
+                    Remapping::into_relative(remapping.unwrap(), &self.root)
+                        .path
+                        .relative()
+                        .join(&import_path.string.replacen(&unwrapped.name.clone(), "", 1))
+                } else {
+                    PathBuf::from(current_path.clone())
+                        .parent()
+                        .unwrap()
+                        .join(import_path.string.clone())
+                };
+
+                let canonical = fs::canonicalize(&remapped).unwrap();
+
+                let sol = fs::read_to_string(&canonical).unwrap_or_else(|_| {
+                    panic!("Could not find file for dependency: {:?}", canonical)
                 });
                 self.file_no += 1;
                 let file_no = self.file_no;
-                let (maybe_entry, mut inner_sources) = self.parse(&sol);
-                inner_sources.push((maybe_entry, path.string.clone(), sol.to_string(), file_no));
+                let (maybe_entry, mut inner_sources) = self.parse(&sol, &remapped);
+                inner_sources.push((
+                    maybe_entry,
+                    remapped.to_str().unwrap().to_owned(),
+                    sol.to_string(),
+                    file_no,
+                ));
                 inner_sources
             }
-            Import::Rename(path, elems, _) => {
+            Import::Rename(import_path, elems, _) => {
                 println!(
-                    "path: {:?}, elems: {:?}, curr: {:?}",
-                    path,
+                    "import_path: {:?}, elems: {:?}, curr: {:?}",
+                    import_path,
                     elems,
                     std::env::current_dir()
                 );
-                let sol = fs::read_to_string(path.string.clone()).unwrap_or_else(|_| {
-                    panic!("Could not find file for dependency: {:?}", path.string)
+                let remapping = self
+                    .remappings
+                    .clone()
+                    .into_iter()
+                    .find(|x| import_path.string.starts_with(&x.name));
+
+                let remapped = if remapping.is_some() {
+                    let unwrapped = remapping.clone().unwrap();
+                    Remapping::into_relative(remapping.unwrap(), &self.root)
+                        .path
+                        .relative()
+                        .join(&import_path.string.replacen(&unwrapped.name.clone(), "", 1))
+                } else {
+                    PathBuf::from(current_path.clone())
+                        .parent()
+                        .unwrap()
+                        .join(import_path.string.clone())
+                };
```

Comment:
similarly this can be simplified as above

Reply:
Applied here as well